### PR TITLE
Fix maxContentLength Issue & Log HttpProtocol used per Service

### PR DESCRIPTION
**Pull request by nizarm**
1. Fix MaxContentLength bug in HttpChannelInitializer
2. Add info log to log information on http protocol used

### CODE REVIEW

File: r2-netty/src/main/java/com/linkedin/r2/netty/client/http/HttpChannelInitializer.java [hunk 1]
```diff
@@ -86,7 +86,7 @@
     _maxInitialLineLength = maxInitialLineLength;
     _maxHeaderSize = maxHeaderSize;
     _maxChunkSize = maxChunkSize;
-    _maxContentLength = maxContentLength > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) maxContentLength;
+    _maxContentLength = maxContentLength;
```
Comment (rachelhanhan) [inline r2-netty/src/main/java/com/linkedin/r2/netty/client/http/HttpChannelInitializer.java:89]: This is in Http2ChannelInitilizer too, should we fix that file as well?

Comment (nizarm) [reply r2-netty/src/main/java/com/linkedin/r2/netty/client/http/HttpChannelInitializer.java:89]: We can't fix it in Http2ChannelInitializer as Netty expects an integer value there and the one in Http2ChannelInitializer is not used in StreamHandler

Comment (nizarm) [reply r2-netty/src/main/java/com/linkedin/r2/netty/client/http/HttpChannelInitializer.java:89]: The one used in Http2ChannelInitializer is used for parent channel and it is not used to serve the same purpose in HttpChannelInitializer. The http2 equivalent of the same can be found @ Http2StreamChannelInitializer

Comment (rachelhanhan) [reply r2-netty/src/main/java/com/linkedin/r2/netty/client/http/HttpChannelInitializer.java:89]: Got it, thanks

**Pull request merged**